source/connect/connection-targets.txt

```diff
@@ -0,0 +1,107 @@
+in your connection string:
+
+- The URL of your Atlas cluster
+- Your MongoDB username
+- Your MongoDB password
+
+Then, pass your connection string to the ``mongoc_client_new()`` function.
+
+.. tip::
+
+   Follow the :atlas:`Atlas driver connection guide </driver-connection>`
+   to retrieve your connection string.
+
+When you connect to Atlas, we recommend using the {+stable-api+} client option to avoid
+breaking changes when Atlas upgrades to a new version of {+mdb-server+}.
+
+.. TODO, uncomment this and link to correct Stable API page for C, once written.
+.. To learn more about the {+stable-api+} feature, see the :ref:`<c-stable-api>`
+guide.
+
+The following code shows how to use the {+driver-short+} to connect to an Atlas cluster. The
+code also uses the ``mongoc_server_api_new()`` function to specify a {+stable-api+} version.
+
+.. literalinclude:: /includes/connect/connection-targets-stable-api.c
+   :language: c
+   :start-after: start-connect
+   :end-before: end-connect
+   :dedent:
+
+Local Deployments
+-----------------
+
+To connect to a local MongoDB deployment, use ``localhost`` as the hostname. By
+default, the ``mongod`` process runs on port 27017, though you can customize this for
+your deployment.
+
+The following code shows how to use the {+driver-short+} to connect to a local MongoDB
+deployment:
+
+.. literalinclude:: /includes/connect/connection-targets-local.c
+   :language: c
+   :start-after: start-connect
+   :end-before: end-connect
+   :dedent:
+
+Replica Sets
+------------
+
+To connect to a replica set, specify the hostnames (or IP addresses) and 
+port numbers of the replica-set members.
+
+If you aren't able to provide a full list of hosts in the replica set, you can 
+specify one or more of the hosts in the replica set and instruct the {+driver-short+} to 
+perform automatic discovery to find the others. To instruct the driver to perform
+automatic discovery, perform one of the following actions:
+
+- Specify the name of the replica set as the value of the ``replicaSet`` parameter.
+- Specify ``false`` as the value of the ``directConnection`` parameter.
+- Specify more than one host in the replica set.
+
+The following example uses a sample connection URI to connect to the MongoDB replica
+set ``myreplset``, which is running on port ``27017`` of three different hosts. 
+
+.. literalinclude:: /includes/connect/connection-targets-replica.c
+   :language: c
+   :start-after: start-connect
+   :end-before: end-connect
+   :dedent:
+
+.. note::
+
+   The ``mongoc_client_new()`` function is *non-blocking*.
+   When you connect to a replica set, the constructor returns immediately while the
+   client uses background threads to connect to the replica set.
+   
+   If you construct a ``mongoc_client_t`` and immediately print the string representation
+   of its ``nodes`` attribute, the list might be empty while the client connects to
+   the replica-set members.
```

Comment:
Suggestion: Add links to API docs for each of the types used on this page. [Example](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/security/enterprise-authentication/#api-documentation) from Python docs.